compiler/dyno/lib/uast/chpl-syntax-printer.cpp

```diff
@@ -767,35 +945,75 @@ struct ChplSyntaxVisitor {
     ss_ << "on ";
     printChapelSyntax(ss_, node->destination());
     ss_ << " ";
-    printBlockWithStyle(node->blockStyle(), node->stmts(), "do ");
+    printBlockWithStyle(node->blockStyle(), node->stmts(), "do ", ";", true);
   }
 
   void visit(const OpCall* node) {
-    // TODO: Where do parens come in?
-    // ex: !(this && that)
-    // is different than !this && that
+    bool needsParens = false;
```

Comment:
I might make two helper functions e.g. `printBinaryOp` and `printUnaryOp` to reduce the branching a bit.